### ISSUE TITLE
Feature Deprecation Life Cycle

### ISSUE THREAD

**Issue by SyntaxNode** [prebid/prebid-server#1619]
Do we already have an approach for how to communicate and implement feature removal? I don't expect this to come up often and I'm asking in the context of https://github.com/prebid/prebid-server/issues/1312.

The change log of PBS-Go is pretty much non-existent beyond commit history (we're fixing this soon with the same approach used by PBS-Java) and even if we had that in a better state right now, do we feel that is that enough communication to remove the feature? i'd like to establish our community decision, document it, and adhere to it when needed in the future.

To get the discussion started...

### Proposal 1
Including the feature deprecation in the change log is sufficient.

### Proposal 2
A phased approach to warning the host:
1. Create a GitHub issue labeled with the 'breaking change' label to discuss the feature being removed and the reasoning. Take no further action until / if it's approved by the Prebid.org Server Committee.
2. Once approved by the Server Committee, add a warning to the startup log if the host has enabled the feature. The deprecation warning will also be included in the change log.
3. Once at least 30 days has passed since releasing the warning, remove the feature.

### Proposal 3
Same as Proposal 2 with intrusive host notices:
1. Create a GitHub issue labeled with the 'breaking change' label to discuss the feature being removed and the reasoning. Take no further action until / if it's approved by the Prebid.org Server Committee.
2. Once approved by the Server Committee, add a warning to the startup log if the host has enabled the feature. The deprecation warning will also be included in the change log.
3. Once at least 30 days has passed since releasing the warning, remove the feature and add a fatal startup error if the host still has it enabled. This will force the host to understand the feature is removed, but runs the risk of being very disruptive to their release process.
4. Once at least 30 days has passed since removing the feature, remove the fatal startup error and ignore the config from this point forward.

Thoughts?

**Comment by SyntaxNode** [comment]
We discussed this at the Prebid Server Committee. We learned that most hosts will carefully review the commit history before accepting, and at the very least verify the changes in a staging environment. We recognize that still may not be enough, so we decided on the following stragety:

1. Create a GitHub issue with the 'breaking change' label discussing the reasoning for removal and alternatives if applicable.
2. Create a PR to emit a warning on application startup.
3. Remove the feature in the following release (typically 1 week) and emit a fatal error on application startup if still configured.
4. After 4 more releases (typically 1 month) remove the fatal error. Feature is now completely removed.
